scripts/bump_deps.lua

```diff
@@ -200,268 +145,86 @@ local function verify_branch(new_branch_suffix)
-  if dependency_name == 'Luv' then
-    write_cmakelists_line(dependency.symbol, 'VERSION', version)
-  end
-  update_cmakelists(dependency, archive, version)
-end
-
-function M.head(dependency_name)
-  local dependency = assert(get_dependency(dependency_name))
-  verify_cmakelists_committed()
-  local commit_sha = get_gh_commit_sha(dependency.repo, 'HEAD')
-  local archive = get_archive_info(dependency.repo, commit_sha)
-  if dependency_name == 'Luv' then
-    warn_luv_symbol()
-  end
-  update_cmakelists(dependency, archive, 'HEAD - ' .. short_commit(commit_sha))
-end
-
-function M.create_branch(dep)
-  explicit_create_branch(dep)
-end
-
-function M.submit_pr()
-  require_executable('git')
-
-  verify_branch('deps')
-
-  local nvim_remote = find_git_remote(nil)
-  local relevant_commit = assert(run_die({
-    'git',
-    'log',
-    '--grep=' .. commit_prefix,
-    '--reverse',
-    "--format='%s'",
-    nvim_remote .. '/master..HEAD',
-    '-1',
-  }, 'Failed to fetch commits'))
-
-  local pr_title
-  local pr_body
-
-  if relevant_commit == '' then
-    pr_title = commit_prefix .. 'bump some dependencies'
-    pr_body = 'bump some dependencies'
-  else
-    relevant_commit = relevant_commit:gsub("'", '')
-    pr_title = relevant_commit
-    pr_body = relevant_commit:gsub(commit_prefix:gsub('%(', '%%('):gsub('%)', '%%)'), '')
-  end
-  pr_body = pr_body .. '\n\n(add explanations if needed)'
-  p(pr_title .. '\n' .. pr_body .. '\n')
-  create_pr(pr_title, pr_body)
+  local commit_sha = get_gh_commit_sha(repo, _ref)
+  local archive = get_archive_info(repo, commit_sha)
+  update_deps_file_wrapper(name, symbol, archive, _ref:sub(1, 9))
 end
 
 local function usage()
-  local this_script = _G.arg[0]:match('[^/]*.lua$')
-  print(([=[
+  local this_script = vim.fs.basename(_G.arg[0])
+  local help = ([=[
     Bump Nvim dependencies
 
     Usage:  nvim -l %s [options]
         Bump to HEAD, tagged version, commit, or branch:
-            nvim -l %s --dep Luv --head
-            nvim -l %s --dep Luv --version 1.43.0-0
-            nvim -l %s --dep Luv --commit abc123
-            nvim -l %s --dep Luv --branch
-        Create a PR:
-            nvim -l %s --pr
+            nvim -l %s --dep luv --head
+            nvim -l %s --dep luv --ref 1.43.0-0
+            nvim -l %s --dep luv --ref abc123
 
     Options:
         -h                    show this message and exit.
-        --pr                  submit pr for bumping deps.
-        --branch <dep>        create a branch bump-<dep> from current branch.
         --dep <dependency>    bump to a specific release or tag.
 
     Dependency Options:
-        --version <tag>       bump to a specific release or tag.
-        --commit <hash>       bump to a specific commit.
-        --HEAD                bump to a current head.
+        --ref <ref>           bump to a specific commit or tag.
+        --head                bump to a current head.
 
         <dependency> is one of:
```

Comment:
I would just add this as a `--list` switch, not bake this into the boilerplate.

Reply:
For bonus points (not in this PR), `--list` could show the latest and tracked versions...